### ISSUE TITLE
Unterminated dollar quote while create PostgreSQL function with SQL-Editor

### ISSUE THREAD

**Issue by Banzai292** [dbeaver/dbeaver#7050]
#### System information: 
- Windows 10 Pro x64 Build 17763
- DBeaver Version 6.2.3.201910211155
- No additional extensions

#### Connection specification:
- PostgreSQL 10
- org.postgresql.Driver
- No tunnels or proxies

#### Describe the problem you're observing:
Executing a loaded script in SQL-Editor with a couple of statements for creating a new schema.
In DBeaver Version 6.2.2 works fine, in 6.2.3 throws error:
`SQL-Error [42601]: Unterminated dollar quote started at position 121 in SQL CREATE OR REPLACE FUNCTION change_logging_trigger()
    RETURNS trigger
    LANGUAGE plpgsql
    SECURITY DEFINER
AS $$
    BEGIN ........`

#### Steps to reproduce, if exist:
Create a function and try different `...AS $xxxx$` variants.

```sql
DROP SCHEMA IF EXISTS logging CASCADE;
CREATE SCHEMA logging;
SET search_path = "logging";
CREATE TABLE t_history (
    id serial,
    datetime timestamp DEFAULT now(),
    schemaname text,
    tablename text,
    operation text,
    modifier text DEFAULT current_user,
    new_val json,
    old_val json,
    CONSTRAINT t_history_pk PRIMARY KEY (id)    
);
CREATE INDEX t_history_datetime_idx ON logging.t_history (datetime);
COMMENT ON TABLE t_history IS 'A history for all actions on triggered actions!';
CREATE OR REPLACE FUNCTION change_logging_trigger()
    RETURNS trigger
    LANGUAGE plpgsql
    SECURITY DEFINER
AS $$
    BEGIN
        if TG_OP = 'INSERT' THEN
            INSERT INTO logging.t_history (schemaname, tablename, operation, new_val)
                VALUES (TG_TABLE_SCHEMA, TG_RELNAME, TG_OP, row_to_json(NEW));
            RETURN NEW;
        ELSIF TG_OP = 'UPDATE' THEN
            INSERT INTO logging.t_history (schemaname, tablename, operation, new_val, old_val)
                VALUES (TG_TABLE_SCHEMA, TG_RELNAME, TG_OP, row_to_json(NEW), row_to_json(OLD));
            RETURN NEW;
        ELSIF TG_OP = 'DELETE' THEN
            INSERT INTO logging.t_history (schemaname, tablename, operation, old_val)
                VALUES (TG_TABLE_SCHEMA, TG_RELNAME, TG_OP, row_to_json(OLD));
            RETURN OLD;
        END IF;
    END;
$$;
```

**Comment by serge-rider** [comment]
Thanks for the bug report. Itt is a side effect of recent $$ strings handle fix.
Workaround: use $$ with tag name for functions create.
E.g. `$function$`

**Comment by Banzai292** [comment]
Sorry, but I try this workaround also (what I mean with "variants of $xxxxx$").
Throws same error.

**Comment by serge-rider** [comment]
Could you try something simple? 
$x$ for instance. Underscores are not supported in tag names yet.

**Comment by pyrocks** [comment]
This bug is the reason I reverted to 6.2.2. All my code uses $$. waiting for a fixed version.

**Comment by Banzai292** [comment]
> 
> 
> Could you try something simple?
> $x$ for instance. Underscores are not supported in tag names yet.

Thx for answer and I gave it a try with `...AS $dowhatyouwant$` and the text turn red. The execution works fine, without error!
So this should be a valid workaround.

Thanks

**Comment by uslss** [comment]
verified

**Comment by neerajbharti** [comment]
> This bug is the reason I reverted to 6.2.2. All my code uses $$. waiting for a fixed version.

Does this mean $$ (without any tag inside two $) will not be supported anymore? I have looootssss of code with $$.

**Comment by JamesInform** [comment]
I have checked the the lastest early access version.

With the last fix the Dollar quoting is basically working.

But when you use $$ then the body of an anonymous code block is not syntax highlighted. Just everything is yellow as a string. When you use something different like e.g. $f$ everything works as exspected.

So
```
do
$$
begin
   raise notice 'hello';
end
$$
;
```

is working but syntax highlighting not.

**Comment by pyrocks** [comment]
I vote for having $$ work as it did pre 6.2.3.

**Comment by ValeTheVioletMote** [comment]
Why was this closed? This remains an issue in 6.3.0!

**Comment by serge-rider** [comment]
This problem is tracked in #7183. 
Please check comments.

**Comment by hermanikal** [comment]
for this issue, in dbeaver 6.2.5 i could solve the issue by use $$ with tag name for functions create. e.g $name_tag$

but, when i download new version of dbeaver (6.3.1) this issue happen again and i cant workaround by add nametag between dollar.

so, i downgrade the version from 6.3.1 to 6.2.5 cause i cant fix this issue in 6.3.1 version
